src/codegen/compile.ml

```diff
@@ -10451,36 +10884,36 @@ and compile_prim_invocation (env : E.t) ae p es at =
       get_val ^^
       G.i (Convert (Wasm.Values.I32 I32Op.WrapI64))
     | Int8, Int16 ->
-      SR.Vanilla,
-      compile_exp_vanilla env ae e ^^
+      SR.UnboxedWord32 Int16,
+      compile_exp_as env ae (SR.UnboxedWord32 Int8) e ^^
       compile_shrS_const 8l
```

Comment:
I think this is the short version of the following logic. The 8 bit shift only works because the tags of `Int8` and `Int16` are identical except that there is a shift by 8 bits. Probably, a more explicit logic would be safer to maintain:
```suggestion
      TaggedSmallWord.lsb_adjust Int8 ^^
      TaggedSmallWord.msb_adjust Int16
```

Reply:
Yep, added commnet